### PR TITLE
en: Add doc about RBAC roles that TiDB Operator requires

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added, or deleted? (Required)

Add doc about RBAC roles that TiDB Operator requires

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-tidb-operator/pull/1205
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by better0332 — APPROVED** [review]
LGTM

File: en/tidb-operator-rbac.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+| persistentvolumes                             | -                 | -              | [get list watch patch update]                    | Perform operations such as adding labels related to cluster information for PV and modifying `persistentVolumeReclaimPolicy` |
+| pods                                          | -                 | -              | [get list watch update delete]                   | Control the access of the Pod resource |
+| nodes                                         | -                 | -              | [get list watch]                                 | Read node labels and set store labels for TiKV and TiFlash accordingly |
+| storageclasses.storage.k8s.io                 | -                 | -              | [get list watch]                                 | Verify whether StorageClass supports `VolumeExpansion` before expanding PVC storage |
+| -                                             |[/metrics]         | -              | [get]                                            | Read monitoring indicators |
+
+> **Note:**
+>
+> * In the **Non-Resource URLs** column, `-` indicates that the item does not have non-resource URLs.
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:67]:
```suggestion
> * In the **Non-resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
```

File: en/tidb-operator-rbac.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+| pods                                          | -                 | -              | [get list watch update delete]                   | Control the access of the Pod resource |
+| nodes                                         | -                 | -              | [get list watch]                                 | Read node labels and set store labels for TiKV and TiFlash accordingly |
+| storageclasses.storage.k8s.io                 | -                 | -              | [get list watch]                                 | Verify whether StorageClass supports `VolumeExpansion` before expanding PVC storage |
+| -                                             |[/metrics]         | -              | [get]                                            | Read monitoring indicators |
+
+> **Note:**
+>
+> * In the **Non-Resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
+> * In the **Resource Names** column, `-` indicates that the item does not have a resource name.
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:68]:
```suggestion
> * In the **Resource name** column, `-` indicates that the item does not have a resource name.
```

File: en/tidb-operator-rbac.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+The following table lists the permissions corresponding to the `tidb-scheduler` ClusterRole.
+
+| Resource                   | Non-Resource URLs | Resource name     | Action                        | Explanation |
+| ---------                  | ----------------- | --------------   | -----                           | ------- |
+| leases.coordination.k8s.io | -                 | -                | [create]                        | Create lease resource locks for leader election |
+| endpoints                  | -                 | -                | [delete get patch update]       | Control the access of the Endpoints resource |
+| persistentvolumeclaims     | -                 | -                | [get list update]               | Read PVC information of PD/TiKV and update the scheduling information to the PVC label |
+| configmaps                 | -                 | -                | [get list watch]                | Read the ConfigMap resource |
+| pods                       | -                 | -                | [get list watch]                | Read pod information |
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:81]:
```suggestion
| pods                       | -                 | -                | [get list watch]                | Read Pod information |
```

File: en/tidb-operator-rbac.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+| nodes                      | -                 | -                | [get list]                      | Read node information |
+| leases.coordination.k8s.io | -                 | [tidb-scheduler] | [get update]                    | Read and update lease resource locks for leader election |
+| tidbclusters.pingcap.com   | -                 | -                | [get]                           | Read Tidbcluster information |
+
+> **Note:**
+>
+> * In the **Non-Resource URLs** column, `-` indicates that the item does not have non-resource URLs.
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:88]:
```suggestion
> * In the **Non-resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
```

File: en/tidb-operator-rbac.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+| configmaps                 | -                 | -                | [get list watch]                | Read the ConfigMap resource |
+| pods                       | -                 | -                | [get list watch]                | Read pod information |
+| nodes                      | -                 | -                | [get list]                      | Read node information |
+| leases.coordination.k8s.io | -                 | [tidb-scheduler] | [get update]                    | Read and update lease resource locks for leader election |
+| tidbclusters.pingcap.com   | -                 | -                | [get]                           | Read Tidbcluster information |
+
+> **Note:**
+>
+> * In the **Non-Resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
+> * In the **Resource Name** column, `-` indicates that the item does not have a resource name.
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:89]:
```suggestion
> * In the **Resource name** column, `-` indicates that the item does not have a resource name.
```

File: en/tidb-operator-rbac.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+## Manage TiDB clusters at the namespace level 
+
+If `clusterScoped=false` is set during the TiDB Operator deployment, TiDB Operator manages TiDB clusters at the Namespace level.
+
+- To check the ClusterRole created for the TiDB Operator, run the following command:
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:95]:
```suggestion
- To check the ClusterRole created for TiDB Operator, run the following command:
```

File: en/tidb-operator-rbac.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    tidb-operator:tidb-controller-manager                                  2021-05-04T13:08:55Z
+    ```
+
+    `tidb-operator:tidb-controller-manager` is the ClusterRole created for the `tidb-controller-manager` Pod.
+
+    > **Note:**
+    >
+    > During the TiDB Operator deployment, if `controllerManager.clusterPermissions.nodes`, `controllerManager.clusterPermissions.persistentvolumes`, `controllerManager.clusterPermissions.storageclasses` are all set to `false`, TiDB operator will not create this ClusterRole.
+
+- To check the roles created for the TiDB Operator, run the following command:
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:115]:
```suggestion
- To check the roles created for TiDB Operator, run the following command:
```

File: en/tidb-operator-rbac.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+    In the output:
+
+    * `tidb-operator:tidb-controller-manager` is the role created for the `tidb-controller-manager` Pod.
+    * `tidb-operator:tidb-scheduler` is the role created for the `tidb-scheduler` Pod.
+
+### `tidb-controller-manager` ClusterRole permissions
+
+The following table lists the permissions corresponding to the `tidb-controller-manager` ClusterRole.
+
+| Resource                   | Non-Resource URLs | Resource name     | Action                        | Explanation |
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:139]:
```suggestion
| Resource                   | Non-resource URLs | Resource name     | Action                        | Explanation |
```

File: en/tidb-operator-rbac.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+| ---------                     | ----------------- | -------------- | -----                            | ------- |
+| persistentvolumes             | -                 | -   | [get list watch patch update]    | Perform operations such as adding labels related to cluster information for PV and modifying `persistentVolumeReclaimPolicy` |
+| nodes                         | -                 | -   | [get list watch]                 | Read node Labels and set store Labels for TiKV and TiFlash accordingly|
+| storageclasses.storage.k8s.io | -                 | -   | [get list watch]                 | Verify whether StorageClass supports `VolumeExpansion` before expanding PVC storage |
+
+> **Note:**
+>
+> * In the **Non-Resource URLs** column, `-` indicates that the item does not have non-resource URLs.
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:147]:
```suggestion
> * In the **Non-resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
```

File: en/tidb-operator-rbac.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+| Resource                   | Non-Resource URLs | Resource name     | Action                        | Explanation |
+| ---------                     | ----------------- | -------------- | -----                            | ------- |
+| persistentvolumes             | -                 | -   | [get list watch patch update]    | Perform operations such as adding labels related to cluster information for PV and modifying `persistentVolumeReclaimPolicy` |
+| nodes                         | -                 | -   | [get list watch]                 | Read node Labels and set store Labels for TiKV and TiFlash accordingly|
+| storageclasses.storage.k8s.io | -                 | -   | [get list watch]                 | Verify whether StorageClass supports `VolumeExpansion` before expanding PVC storage |
+
+> **Note:**
+>
+> * In the **Non-Resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
+> * In the **Resource Name** column, `-` indicates that the item does not have a resource name.
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:148]:
```suggestion
> * In the **Resource name** column, `-` indicates that the item does not have a resource name.
```

File: en/tidb-operator-rbac.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+
+### `tidb-controller-manager` Role permissions
+
+The following table lists the permissions corresponding to the `tidb-controller-manager` Role.
+
+| Resource                                      | Non-Resource URLs | Resource name     | Action                        | Explanation |
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:154]: ditto

File: en/tidb-operator-rbac.md [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+| roles.rbac.authorization.k8s.io               | -                 | -              | [escalate create get update delete]              | Create Role for the TidbMonitor/Discovery service |
+| persistentvolumeclaims                        | -                 | -              | [get list watch create update delete patch]      | Control the access of the PVC resource |
+| jobs.batch                                    | -                 | -              | [get list watch create update delete]            | Use jobs to perform TiDB cluster initialization, backup, and restore operations |
+| pods                                          | -                 | -              | [get list watch update delete]                   | Control the access of the Pod resource |
+
+> **Note:**
+>
+> * In the **Non-Resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
+> * In the **Resource Names** column, `-` indicates that the item does not have a resource name.
+> * In the **Actions** column, `*` indicates that the resource supports all actions that can be performed on a Kubernetes cluster.
```
Comment (TomShawn) [inline en/tidb-operator-rbac.md:179]: ditto
Use sentence capitalization for the table headers consistently.

File: en/tidb-operator-rbac.md [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+| secrets                                       | -                 | -              | [create update get list watch delete]            | Control the access of the Secret resource |
+| roles.rbac.authorization.k8s.io               | -                 | -              | [escalate create get update delete]              | Create Role for the TidbMonitor/Discovery service |
+| persistentvolumeclaims                        | -                 | -              | [get list watch create update delete patch]      | Control the access of the PVC resource |
+| jobs.batch                                    | -                 | -              | [get list watch create update delete]            | Use jobs to perform TiDB cluster initialization, backup, and restore operations |
+| pods                                          | -                 | -              | [get list watch update delete]                   | Control the access of the Pod resource |
+
+> **Note:**
+>
+> * In the **Non-Resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
+> * In the **Resource Names** column, `-` indicates that the item does not have a resource name.
```
Comment (qiancai) [inline en/tidb-operator-rbac.md:178]:
```suggestion
> * In the **Non-resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
> * In the **Resource names** column, `-` indicates that the item does not have a resource name.
```

File: en/tidb-operator-rbac.md [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+> * In the **Actions** column, `*` indicates that the resource supports all actions that can be performed on a Kubernetes cluster.
+
+### `tidb-scheduler` Role permissions
+
+The following table lists the permissions corresponding to the `tidb-scheduler` Role.
+
+| Resource                   | Non-Resource URLs | Resource name     | Action                        | Explanation |
```
Comment (qiancai) [inline en/tidb-operator-rbac.md:185]:
```suggestion
| Resource                   | Non-resource URLs | Resource name     | Action                        | Explanation |
```

File: en/tidb-operator-rbac.md (hunk 11 above)
Comment (qiancai) [reply en/tidb-operator-rbac.md:154]:
```suggestion
| Resource                                      | Non-resource URLs | Resource name     | Action                        | Explanation |
```

File: en/tidb-operator-rbac.md [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+>
+> * In the **Non-resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
+> * In the **Resource name** column, `-` indicates that the item does not have a resource name.
+> * In the **Actions** column, `*` indicates that the resource supports all actions that can be performed on a Kubernetes cluster.
+
+### `tidb-scheduler` ClusterRole permissions
+
+The following table lists the permissions corresponding to the `tidb-scheduler` ClusterRole.
+
+| Resource                   | Non-Resource URLs | Resource name     | Action                        | Explanation |
```
Comment (qiancai) [inline en/tidb-operator-rbac.md:75]:
```suggestion
| Resource                   | Non-resource URLs | Resource name     | Action                        | Explanation |
```

File: en/tidb-operator-rbac.md [hunk 16]
```diff
@@ -0,0 +1,199 @@
...
+In the output above:
+
+* `tidb-operator:tidb-controller-manager` is the ClusterRole created for the `tidb-controller-manager` Pod.
+* `tidb-operator:tidb-scheduler` is the ClusterRole created for the `tidb-scheduler` Pod.
+
+### `tidb-controller-manager` ClusterRole permissions
+
+The following table lists the permissions corresponding to the `tidb-controller-manager` ClusterRole.
+
+| Resource       | Non-Resource URLs        | Resource name          | Action                        | Explanation |
```
Comment (qiancai) [inline en/tidb-operator-rbac.md:38]:
```suggestion
| Resource       | Non-resource URLs        | Resource name          | Action                        | Explanation |
```

File: en/tidb-operator-rbac.md [hunk 17]
```diff
@@ -0,0 +1,199 @@
...
+| configmaps                 | -                 | -                | [get list watch]                | Read the ConfigMap resource |
+| pods                       | -                 | -                | [get list watch]                | Read pod information  |
+| nodes                      | -                 | -                | [get list]                      | Read node information |
+| leases.coordination.k8s.io | -                 | [tidb-scheduler] | [get update]                    | Read and update lease resource locks for leader election |
+| tidbclusters.pingcap.com   | -                 | -                | [get]                           | Read Tidbcluster information |
+
+> **Note:**
+>
+> * In the **Non-Resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
+> * In the **Resource Name** column, `-` indicates that the item does not have a resource name.
```
Comment (qiancai) [inline en/tidb-operator-rbac.md:199]:
```suggestion
> * In the **Non-resource URLs** column, `-` indicates that the item does not have non-resource URLs. 
> * In the **Resource name** column, `-` indicates that the item does not have a resource name.
```

**Comment by qiancai** [comment]
@TomShawn Thanks for the review. I've applied all your comments. PTAL again.

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**